src/corelibs/U2Gui/src/util/project/ProjectUpdater.cpp

```diff
@@ -135,17 +135,19 @@ void ProjectUpdater::fetchObjectsInUse(const U2DbiRef &dbiRef, U2OpStatus &os) {
 void ProjectUpdater::updateAccessedObjects() {
     const QList<GObjectViewWindow *> activeViews = GObjectViewUtils::getAllActiveViews();
     QMap<U2DbiRef, DbiConnection *> dbiRef2Connections;  // when changing the code below, beware mem leaks
-    U2OpStatus2Log os;
     foreach (GObjectViewWindow *view, activeViews) {
         foreach (GObject *object, view->getObjects()) {
+            U2OpStatus2Log os;
             Document *doc = object->getDocument();
             if (nullptr != doc && doc->isStateLocked()) {
                 continue;
             }
             const U2EntityRef ref = object->getEntityRef();
             if (!dbiRef2Connections.contains(ref.dbiRef)) {
-                dbiRef2Connections.insert(ref.dbiRef, new DbiConnection(ref.dbiRef, os));
+                auto dbiCon = new DbiConnection(ref.dbiRef, os);
                 CHECK_CONTINUE(!os.hasError());
```

Comment:
C++ is not a memory managed language. Please think about memory deallocation every time you create objects with "new"

Reply:
I could make this pointer shared, but the point is, that if `os` has an error, that means, that the memory hasn't been allocated and there's nothing to free. Do you want me to use `QSharedPointer`?

Reply:
CHECK is an expected behavior and should be processed correctly: you can manually delete the pointer or use a shared pointer: once there are no memory leaks both solutions are good.

As an alternative you can use SAFE_POINT: in this case we can allow memory leaks because we do not expect this behavior at all and can have non critical memory leaks like this.

Up to you, I'm OK with both options.

Reply:
OK, shared pointers are used

Reply:
May be there is a better solution here (optional, up to you) : the problem is that we use a constructor to make some logic and want to check the result value. 

A classic factory method will do a better job in this case: use a static factory method instead of the constructor to instantiate the object.

The factory method should return 'nullptr' if instantiation is failed, so no memory management is needed on the caller side, it is all done in the callee (factory method)

Reply:
It's a solution, of course, but I prefer to use shared pointers where it's possible cuz this looks like a more advanced way of resolving the memory allocation problems - we do not need to think about it at all, the smart pointer do it for us. So, if it's not necessary, I'd rather leave `QSharedPointer` here